### DISCUSSION TITLE
Sharing data for independent testing

### DISCUSSION THREAD

**Discussion by abhi18av** [TORCH-Consortium/MAGMA#20] [Q&A] [answered]
Hi @TimHHH ,

I was wondering whether it'd be possible for you to share with me the data such as resources and reference files? 

This way I can test the pipeline from end to end independently, preferably on 5 genome samples, and make sure that the results are as expected. 

If possible, could you can share the data via OneDrive

**Comment by TimHHH** [comment]
Hi @abhi18av ,
I believe all the resources can now be found in the ```resources``` branch. I am assuming that you also require the fastqs, I will get these to you via OneDrive. Note that I will supply a couple more samples to ensure that VQSR behaves reasonably well.

**Comment by TimHHH** [comment, marked as answer]
You can find the fastqs here:
http://transfer.sh/qC7cUo/M_16_F0425_EC550A-330_Myco-F5_Mycobacterium_S32_L005_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0425_EC550A-330_Myco-F5_Mycobacterium_S32_L005_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0458_FS015-12_Myco-F5_Mycobacterium_S65_L005_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0458_FS015-12_Myco-F5_Mycobacterium_S65_L005_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0465_FS021A-289_Myco-F5_Mycobacterium_S72_L005_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0465_FS021A-289_Myco-F5_Mycobacterium_S72_L005_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0467_FS023-73_Myco-F5_Mycobacterium_S74_L005_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0467_FS023-73_Myco-F5_Mycobacterium_S74_L005_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0586_EC012B-270_Myco_F-7_Mycobacterium_ACCACTGT_L008_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0586_EC012B-270_Myco_F-7_Mycobacterium_ACCACTGT_L008_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0644_EC083-202_Myco_F-7_Mycobacterium_TCTTCACA_L008_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0644_EC083-202_Myco_F-7_Mycobacterium_TCTTCACA_L008_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0649_EC088-207_Myco_F-7_Mycobacterium_TTCACGCA_L008_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0649_EC088-207_Myco_F-7_Mycobacterium_TTCACGCA_L008_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0650_EC090-209_Myco_F-7_Mycobacterium_AACTCACC_L008_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_F0650_EC090-209_Myco_F-7_Mycobacterium_AACTCACC_L008_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_J0666_FS007_2_85_NFHT58_L006_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_J0666_FS007_2_85_NFHT58_L006_R2_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_J0673_fs031_2_89_NFHT86_L006_R1_001.fastq.gz
http://transfer.sh/qC7cUo/M_16_J0673_fs031_2_89_NFHT86_L006_R2_001.fastq.gz
http://transfer.sh/qC7cUo/xbs-nf.test.csv
Note the new and improved sample sheet, this one can be included in the development branch.
Next time I'll combine the lot in one zip 😉 first time using transfer.sh.

**Reply by abhi18av** [reply]
This sounds great @TimHHH , appreciate it!

**Reply by abhi18av** [reply]
I've downloaded the files from there now.

**Reply by abhi18av** [reply]
Hi @TimHHH  @LennertVerboven , 

Could you please share these FASTQ files with me again? I have lost access to the server I was using to test these, I might be able to access it again by tomorrow - but just in case, I'd rather have the data again as a backup.

**Reply by TimHHH** [reply]
Hi @abhi18av 
Do you still need these files? Or have you managed to find a copy?

**Reply by abhi18av** [reply]
Yup, found a copy on my end - thanks!

**Comment by abhi18av** [comment]
I think that there are a couple resource remaining 🤔 


**1.** `resistance_db` 

https://github.com/abhi18av/xbs-nf/blob/07c8b73e037f89a2f821ff6d4e935e91b05d3ea8/conf/dev.config#L52

This is from the original, XBS_main.py (Line-79)

```python
	parser.add_argument('--resistance_db', metavar='resistance_db', type=str, default='resistance_db_v1', help='The database to use for resistance calling')
```


**2.** Is this file https://github.com/abhi18av/xbs-nf/blob/master/resources/regions/UVP_List_of_Excluded_loci.list the same as `DRgenes.list` which is mentioned in XBS_call.sh (Line-140) ?


**3.** Similarly, there are 3 vcf files which I can't find.

```nextflow

    ref_gvcf="${projectDir}/data/test_data/reference_gvcfs_v02.g.vcf.gz"

    ref_gvcf_study=""

    dbsnp_vcf = "${projectDir}/data/test_data/tbvar.v3.vcf"


```



Or perhaps I've overlooked these and they were already included as part of https://github.com/abhi18av/xbs-nf/pull/23 albeit named differently?

**Reply by TimHHH** [reply]
> **1.** `resistance_db`

We have used two versions of this db, the original (?) TBprofiler version and Lennert's WHO version. @LennertVerboven  can you suggest which version is best to use?



> **2.** Is this file https://github.com/abhi18av/xbs-nf/blob/master/resources/regions/UVP_List_of_Excluded_loci.list the same as `DRgenes.list` which is mentioned in XBS_call.sh (Line-140) ?

These are different, I still have to create `DRgenes.list` but this depends on which `resistance_db` we use @LennertVerboven 


> **3.** Similarly, there are 3 vcf files which I can't find.

The are old redundant files. Code can be scrapped.

**Reply by abhi18av** [reply]
@TimHHH @LennertVerboven 

Revisiting this on today again to test alternative pathways within the pipeline, since `dbsnp` is a redundant file, used in the `CALL_WF` 

```nextflow
        if (params.dataset_is_not_contaminated) {
            // call_base_recal
            GATK_BASE_RECALIBRATOR(GATK_MARK_DUPLICATES.out.bam_tuple, params.dbsnp_vcf, params.ref_fasta)

            // call_apply_bqsr
            GATK_APPLY_BQSR(GATK_BASE_RECALIBRATOR.out, params.ref_fasta)


            recalibrated_bam_ch = GATK_APPLY_BQSR.out

        } else {

            recalibrated_bam_ch = GATK_MARK_DUPLICATES.out.bam_tuple
        }


```

Since the file is redundant, then shall I trim the `if` condition and only keep the following? 

```nextflow
            recalibrated_bam_ch = GATK_MARK_DUPLICATES.out.bam_tuple
```

**Reply by abhi18av** [reply]
The `GATK_BASE_RECALIBRATOR` process, doesn't seem to have been used anywhere else.

**Reply by TimHHH** [reply]
Hi @abhi18av 
Actually I like this piece of code and it would be nice to keep. You can simply use [Benavente2015](https://github.com/abhi18av/xbs-nf/blob/master/resources/known/Benavente2015.UVPapproved.rRNAexcluded.vcf.gz) as the ```dbsnp```.
This piece of code should not be default and only activated with some argument stating that the dataset is uncontaminated, but I think this has already been implemented?

**Reply by abhi18av** [reply]
Yes, this is already implemented and is only activated when the `dataset_is_not_contaminated` is marked as `true`.

However, since there is no `dbsnp_vcf` it fails currently. Based on your comment, I'll add `Benavente2015.UVPapproved.rRNAexcluded.vcf.gz` as the default one - or do you wish to leave it to the user to use within their config?

**Reply by TimHHH** [reply]
```Benavente2015.UVPapproved.rRNAexcluded.vcf.gz``` can be default.

**Reply by abhi18av** [reply]
Oki doki, I've made that one as the default.

https://github.com/abhi18av/xbs-nf/pull/46/commits/dfd932b76aa8d922666160b6b93083064f48057e

**Comment by LennertVerboven** [comment]
For the first point; these databases have to be loaded in TBProfiler, this needs to happen during install using some reference files and a list of variants and their association. I can look up which files are required and send them to you.


Lennert Verboven



[logo_uantwerp] <https://www.uantwerp.be>    [logo_torch]  <https://torch-consortium.com/>

        TORCH - Tuberculosis Omics ResearCH consortium
Global Health Institute
Gouverneur Kinsbergencentrum - Room 1.16
Doornstraat 331 - 2610 Wilrijk - Belgium
***@***.*** ***@***.***>

________________________________

**Reply by TimHHH** [reply]
is the WHO database now in a usable format or do you prefer to stick to the standard TBprofilerdb? did you also not supplement the TB standard TBprofiler db?

**Comment by LennertVerboven** [comment]
WHO is not yes usable, the TBprofiler db is installed by default, Imade some additions, I will supply the required files and the the commands to load the db. You can always omit the db paramter and it will use the default TBProfiler one for testin and development.


Lennert Verboven



[logo_uantwerp] <https://www.uantwerp.be>    [logo_torch]  <https://torch-consortium.com/>

        TORCH - Tuberculosis Omics ResearCH consortium
Global Health Institute
Gouverneur Kinsbergencentrum - Room 1.16
Doornstraat 331 - 2610 Wilrijk - Belgium
***@***.*** ***@***.***>

________________________________

**Reply by abhi18av** [reply]
> You can always omit the db paramter and it will use the default TBProfiler one for testin and development.

Ah, this is the crucial bit - it means this needs to be exposed as an optional database (file).

Thanks for letting me know @LennertVerboven 👍

**Reply by TimHHH** [reply]
Hi.
@abhi18av the original code speaks of a `resistance_db` which is normally loaded from the tbprofiler venv (a set of files all starting with the same name). Would be great to have the possibility to easily specify different db's as we anticipate updated versions continuously.
When @LennertVerboven supplies the resistance_db this will include a `.bed` file that can be used as the `DRgenes.list` file, this also ensures that the DR genes match those from the database being specified.

**Reply by abhi18av** [reply]
Hi @TimHHH ,

> Would be great to have the possibility to easily specify different db's as we anticipate updated versions continuously.

Sure, this makes sense and should be possible using the existing design of passing this as params for example, in the `pbs.config` you can specify this 

https://github.com/abhi18av/xbs-nf/blob/07c8b73e037f89a2f821ff6d4e935e91b05d3ea8/conf/dev.config#L52

When you wish to use a different version of the `resistance_db_v100` then you'd simply need to update the `pbs.config` file for the same 

```nextflow
 resistance_db="${projectDir}/data/mock_data/resistance_db_v100" 
``` 

Does this suffice the use-case?

Though, it must be kept in mind that if the `resistance_db` is updated on an analysis which you want to resume (after adding more data) - this would not be possible, since the `md5sum` of the inputs of the process would be different and Nextflow would re-evaluate all samples within that process since one of the input files i.e. `resistance_db` changed.


> ... supplies the resistance_db this will include a .bed file that can be used as the DRgenes.list file, this also ensures that the DR genes match those from the database being specified.

Sure, that would be great to have to finish the testing on my end 👍

**Reply by TimHHH** [reply]
okay, that sounds good I think.
